### PR TITLE
Support for dynamic MQTT user configuration #4883

### PR DESCRIPTION
**Pull request by baylanger**
This should be enough, otherwise let me know.

### CODE REVIEW

File: docs/docs/configuration/index.md [hunk 1]
```diff
@@ -53,6 +53,9 @@ mqtt:
   client_id: frigate
   # Optional: user
   user: mqtt_user
+  # NOTE: MQTT user can be specified with an environment variables that must begin with 'FRIGATE_'.
```
Comment (NickM-27) [inline docs/docs/configuration/index.md:56]:
```suggestion
   client_id: frigate
   # Optional: user
  # NOTE: MQTT user can be specified with an environment variables that must begin with 'FRIGATE_'.
   user: mqtt_user
```

File: frigate/config.py [hunk 2]
```diff
@@ -866,6 +866,9 @@ def runtime_config(self) -> FrigateConfig:
         # MQTT password substitution
         if config.mqtt.password:
             config.mqtt.password = config.mqtt.password.format(**FRIGATE_ENV_VARS)
+        # MQTT user substitution
+        if config.mqtt.user:
+            config.mqtt.user = config.mqtt.user.format(**FRIGATE_ENV_VARS)
```
Comment (NickM-27) [inline frigate/config.py:871]:
```suggestion
         if config.mqtt.user or config.mqtt.password:
             config.mqtt.user = config.mqtt.user.format(**FRIGATE_ENV_VARS)
             config.mqtt.password = config.mqtt.password.format(**FRIGATE_ENV_VARS)
```

Comment (NickM-27) [reply frigate/config.py:871]: User should be able to be substituted even if password isn't set

File: docs/docs/configuration/index.md (hunk 1 above)
Comment (NickM-27) [reply docs/docs/configuration/index.md:56]: Comment needs to be above the value.

File: docs/docs/configuration/index.md [hunk 3]
```diff
@@ -53,6 +53,9 @@ mqtt:
...
+  #       e.g. password: '{FRIGATE_MQTT_USER}'
```
Comment (baylanger) [inline docs/docs/configuration/index.md:57]: Almost blind copy/paste, will fix.

**Review by NickM-27 — CHANGES_REQUESTED** [review]

**Review by NickM-27 — APPROVED** [review]

File: docs/docs/configuration/index.md (hunk 1 above)
Comment (baylanger) [reply docs/docs/configuration/index.md]: Fixed in latest commit, had not pay attn enough.

**Pull request merged**